### PR TITLE
fixes for MODULES-1255, MODULES-1253, MODULES-1252, MODULES-1250

### PR DESCRIPTION
**Pull request by cristifalcas**
fixes for MODULES-1255, MODULES-1253, MODULES-1252, MODULES-1250

### CODE REVIEW

**Comment by maclennann** [comment]
Is there anything preventing this from being merged? I need to be able to use the config merging to turn off base64, but duplicate resources are stopping me from doing so.

**Comment by daenney** [comment]
There's a couple of things:

* This should be separate commits, preferably even separate PRs
* It's lacking tests covering the new behaviour

File: templates/refresh-mcollective-metadata.erb [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 require 'facter'
 require 'facter/application'
+<% if @osfamily=='RedHat' then -%>
```
Comment (igalic) [inline templates/refresh-mcollective-metadata.erb]: might wanna add Amazon, too?

**Comment by jhg03a** [comment]
We should now be able to close this PR

**Pull request closed without merging**